tools/asset-sync/assets.Tests.Helpers.ps1

```diff
@@ -0,0 +1,208 @@
+# This powershell script contains multiple functions that are dot-included in the BeforeAll function of assets.Tests.ps1
+#
+# Describe-TestFolder is the primary entrypoint, and is used to create a test environment given some basic setup input. This function is used to:
+#  - Create faux recording.jsons and place them in a fake language repo of any structure
+#  - Create any fake language repo structure with some random sample data
+#  - Dynamically create a copy of the assets repo within a local folder to actually create real test scenarios
+#  - Will push duplicates of test scenario branches to alternative, guid-fied branch names for actual push/pull operations.
+
+Function Get-Basic-AssetsJson {
+  param(
+    [boolean]$RandomizeRepoId = $true
+  )
+  $result = (Get-Content -Raw -Path (Join-Path $PSScriptRoot "assets.json") | ConvertFrom-Json)
+
+  if ($RandomizeRepoId) {
+    Add-Member -InputObject $result -MemberType "NoteProperty" -Name "AssetRepoId" -Value (New-Guid).ToString()
+  }
+
+  return $result
+}
+
+Function Get-Full-TestPath {
+  Param(
+      [string] $Path
+  )
+  return $Path.Replace('TestDrive:', (Get-PSDrive TestDrive).Root)
+}
+
+Function Get-TestFolder {
+  return (Join-Path $PSScriptRoot ".testruns")
+}
+
+Function Get-TestPath {
+  $disablePermanentStorage = $env:USE_TESTDRIVE
+
+  $localTempPath = Get-TestFolder
+
+  if ($disablePermanentStorage){
+    $testPath = "TestDrive:\$([Guid]::NewGuid())\"
+    
+    mkdir $testPath | Out-Null
+    return $testPath
+  }
+  else {
+    $testPath = (Join-Path $localTempPath "$([Guid]::NewGuid())")
+
+    mkdir -p $testPath | Out-Null
```

Comment:
Not guaranteed to work on windows if you don't have git bash/pwsh installed. `New-Item -Type Directory -Force`